### PR TITLE
[DSY-1515]Fix/maintain icon codes

### PR DESCRIPTION
**Pull request by arielwb**
# Description

This PR fixes the change of icon codes when a new icon is included in the lib.

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [x] Unit Test
- [x] Output test
- [x] Regression test

Below a comparison between two regression tests adding the same icon, before and after the fix

before the fix

before-fix

after the fix 

after-fix

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: packages/natds-icons/src/index.js [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+  .then(buildInitialFontData)
+  .then(buildCurrentIconCodes)
+  .then(buildJson)
+  .then(buildJsonDts)
+  .then(buildCss)
+  .then(buildCssDts)
+  .then(buildFonts)
+  .then(buildSwift)
+  .then(writeFiles)
+  .catch(log);
```
Comment (jimmyandrade) [inline packages/natds-icons/src/index.js:40]: Muito chique!

File: packages/natds-icons/integration/__file_snapshots__/natds-icons.snap.json.d.ts [hunk 2]
```diff
@@ -1,4 +1,3 @@
-/* eslint-disable max-lines,quotes */
```
Comment (jimmyandrade) [inline packages/natds-icons/integration/__file_snapshots__/natds-icons.snap.json.d.ts:1]: Eu lembro que tive de colocar isso quando ficava "do lado de cá" porque eu não conseguia ajustar as aspas haha

**Review by jimmyandrade — APPROVED** [review]
LGTM, só deixei uma pergunta

File: packages/natds-icons/dist/index.d.ts [hunk 3]
```diff
@@ -0,0 +1,8 @@
+import * as iconStyles from './natds-icons.css';
```
[the remark this answers is not in the document]
Comment (arielwb) [reply packages/natds-icons/dist/index.d.ts:1]: pq a gente já comitava os arquivos gerados na pasta src, e manter duas pastas pros outputs dificulta a vida

File: packages/natds-icons/src/actions/writeFiles.test.js [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+    path.dirname.mockReturnValue('folder/name');
+    fs.existsSync.mockReturnValue(false);
+
+    writeFiles(data);
+
+    expect(fs.existsSync).toHaveBeenCalled();
+    expect(fs.mkdirSync).toHaveBeenCalledWith('folder/name', { recursive: true });
+  });
+
+  it('should not crete the output directory if alreay exists', () => {
```
Comment (jimmyandrade) [inline packages/natds-icons/src/actions/writeFiles.test.js:43]: Typo: aqui seria `create` em vez de `crete`, mas não é bloqueante :)

**Review by jimmyandrade — APPROVED** [review]
LGTM

File: packages/natds-icons/src/actions/writeFiles.test.js (hunk 4 above)
Comment (arielwb) [reply packages/natds-icons/src/actions/writeFiles.test.js]: nice catch! ;)

**Pull request merged**